feat: setup emotion styling lib

kleviss opened the pull request:
feat: Implemented consistent theming using Emotion's styled components
- Refactored multiple screens and components using Emotion styled components
- Updated theme constants with more comprehensive color palette
- Added logo image and improved styling for authentication, home, and car detail screens
- Enhanced bottom sheet and filter components for better user interaction
- Refactored screens to better reflect their naming

Marlen0307 commented on app/(auth)/sign-in.tsx:
```diff
@@ -27,81 +73,48 @@ export default function SignIn() {
-        style={[styles.input, { color: theme.colors.text }]}
+      {error && (
+        <View style={{ marginBottom: 15 }}>
+          <StyledText variant="error" theme={theme}>{error}</StyledText>
```
nqs eshte ThemeProvider siper ketyre componenteve nuk ke nevoj tja kalosh `theme` ne menyre explicite apo jo? Kam pershtypjen qe kane te gjithe comonentet access

kleviss replied:
Let me try

Marlen0307 commented on app/(auth)/sign-up.tsx:
```diff
@@ -29,88 +74,54 @@ export default function SignUp() {
-      <Text style={styles.title}>Create Account</Text>
+    <StyledContainer>
+      <Image
+        source={require('../../assets/images/logo.png')}
```
ka ndonje arsye pse ke perdorur require? zakonisht `import` on top of the file should work per te qene me konsistent

kleviss replied:
Ska arsye gango, ashtu e kishte projketi starter, ja ta prv njeher me import

kleviss replied:
@Marlen0307 - Bera nje mini research, dhe ca ndryshime tek commiti rradhes, por qe referohu ktij SO thread for more: https://stackoverflow.com/a/69995694/10968401

Marlen0307 commented on package.json:
```diff
@@ -41,6 +43,7 @@
     "react-dom": "18.3.1",
     "react-native": "0.76.6",
     "react-native-async-storage": "^0.0.1",
+    "react-native-color-matrix-image-filters": "^7.0.2",
```
per cfare eshte kjo dependency

kleviss replied:
Opps, unused package, srry!!

kleviss committed:
refactor: clean up sign-in and sign-up components

- Remove unused imports and commented-out code
- Simplify styled components with TypeScript generics
- Create reusable StyledImage component
- Improve type safety and reduce redundancy in styling
- Remove inline image styling in favor of styled components

kleviss committed:
Merge branch 'master' into feat/setup-emotion-styling-lib

feat: integrate React Query and enhance search screen functionality

- Added React Query for state management and data fetching
- Integrated @tanstack/react-query with DevTools plugin
- Implemented network state listener for online/offline query management
- Refactored search screen to use FlashList for performance
- Created new SearchPageHeader component for better filter management
- Added new public and protected route layouts
- Restructured protected screens with nested routing
- Added placeholder screens for various app sections
- Updated dependencies to support new features
